app/web/src/components/ModelingDiagram/diagram_types.ts

```diff
@@ -98,45 +96,45 @@ export class DiagramEdgeData extends DiagramElementData {
 
   // helpers to get the unique key of the node and sockets this edge is connected to
   get fromNodeKey() {
-    const comp = useComponentsStore().componentsByNodeId[this.def.fromNodeId];
+    const comp = useComponentsStore().componentsById[this.def.fromComponentId];
     if (comp?.isGroup) {
-      return DiagramGroupData.generateUniqueKey(this.def.fromNodeId);
+      return DiagramGroupData.generateUniqueKey(this.def.fromComponentId);
     }
-    return DiagramNodeData.generateUniqueKey(this.def.fromNodeId);
+    return DiagramNodeData.generateUniqueKey(this.def.fromComponentId);
   }
 
   get toNodeKey() {
-    const comp = useComponentsStore().componentsByNodeId[this.def.toNodeId];
+    const comp = useComponentsStore().componentsById[this.def.toComponentId];
     if (comp?.isGroup) {
-      return DiagramGroupData.generateUniqueKey(this.def.toNodeId);
+      return DiagramGroupData.generateUniqueKey(this.def.toComponentId);
     }
-    return DiagramNodeData.generateUniqueKey(this.def.toNodeId);
+    return DiagramNodeData.generateUniqueKey(this.def.toComponentId);
```

Comment:
is it worth replacing these unique keys with the component pks?

Reply:
I don't think so. If we are in a changeset, any unchanged components will just refer to the entry on head, so they'll have a different PK, but the same ID. The uniqueness here is among different components and pk distinguishes amongst "versions"  of components too